romancal/skycell/match.py

```diff
@@ -0,0 +1,173 @@
+"""
+This module determines which sky cells overlap with the given image.
+
+Currently this assumes that the sky projected borders of all calibrated L2 images are straight.
+"""
+
+import logging
+from functools import cached_property
+
+import numpy as np
+import spherical_geometry.polygon as sgp
+import spherical_geometry.vector as sgv
+from gwcs import WCS
+from numpy.typing import NDArray
+
+import romancal.skycell.skymap as sm
+from romancal.assign_wcs.utils import wcs_bbox_from_shape
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+class ImageFootprint:
+    """abstraction of an image footprint"""
+
+    __radec_corners: NDArray[float]
+
+    def __init__(self, radec_corners: list[tuple[float, float]]):
+        """
+        Parameters
+        ----------
+        radec_corners: list[tuple[float, float]]
+            four corners of the image in right ascension and declination
+        """
+        radec_corners = np.array(radec_corners)
+        if radec_corners.shape[0] != 4:
+            raise ValueError(f"need 4 corners, not {radec_corners.shape[0]}")
+        self.__radec_corners = radec_corners
```

Comment:
I have the impression (perhaps mistaken) that not many use the double underscore private attribute scheme in python, partly because it can be bypassed easily. The convention generally seems to be to use a single underscore to indicate it is private, and if you want to violate it by accessing it elsewhere, go ahead, but "you were warned"

Reply:
In this case, this attribute should never be modified, since it would misalign the cached properties relying on it. It doesn't make much sense to modify it after creating the object either, at least I can't think of a use case where that would be needed instead of just making a new object

Reply:
If the goal is to prevent modifying the attribute, the usual approach is to make it a property without a setter. The main difference between `self.__radec_corners` and `self._radec_corners` is that the latter indicates that the attribute is private and relies on convention not to use it, where as the former is a clumsy way of making inaccessible to code outside the class, though it is a very simple manner to access it by understanding how Python is mangling the name. My impression is few rely on this. Instead you can expose the value as a property but not provide a setter so no one can assign a new value to the property.

Reply:
well I still don't see much functional difference between the two, but it's a trivial change so I'll update the classes

Reply:
c07a61fc052ed039a2c34a3d4c097391b5a66085

Reply:
It's not so much the functional difference than the fact when it is little used it may confuse other maintainers. It's basically a style consistency issue.